hermes/bin/tests/integration/components/sleep_component/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+//! The test Hermes App.
+#![allow(
+    clippy::missing_safety_doc,
+    clippy::missing_docs_in_private_items,
+    clippy::expect_used
+)]
+
+mod bindings {
+
+    wit_bindgen::generate!({
+        world: "hermes",
+        path: "../../../../../../wasm/wasi/wit",
+        generate_all,
+    });
+}
+mod stub;
+mod utils;
+
+use std::{
+    fs,
+    io::{Seek as _, SeekFrom, Write as _},
+};
+
+use crate::utils::{busy_wait_s, make_payload, test_log};
+
+const REQUEST_COUNT: usize = 5;
+const RESPONSES_FILE: &str = "responses.txt";
+const CONTENT: &[u8] = b"\xF0\x9F\xA6\x80\n";
+const WAIT_FOR_SECS: u64 = 1;
+
+struct HttpRequestApp;
+
+impl bindings::exports::hermes::init::event::Guest for HttpRequestApp {
+    fn init() -> bool {
+        test_log("init sleep component");
+        let settings_json = fs::read_to_string("/lib/sleep_module/settings.schema.json")
+            .expect("cannot read settings file");
+        let parsed_json: serde_json::Value =
+            serde_json::from_str(&settings_json).expect("unable to parse settings as JSON");
+        let http_server = parsed_json
+            .get("http_server")
+            .expect("missing http_server in settings")
+            .as_str()
+            .expect("http_server is not a string");
+
+        let mut file = std::fs::File::create(RESPONSES_FILE).expect("failed to create file");
```

Comment:
Its not a great idea to use the filesystem for this because of the issues you noted, but in any event, even using os files this way is problematic without file locking.  I suggest instead use the sqlite DB to keep whatever records, because it will properly handle contention and simultaneous updates/accesses.